cpp/src/arrow/parquet/parquet-io-test.cc

```diff
@@ -52,25 +53,104 @@ template <typename TestType>
 struct test_traits {};
 
 template <>
+struct test_traits<BooleanType> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::BOOLEAN;
+  static constexpr LogicalType::type logical_enum = LogicalType::NONE;
+  static uint8_t const value;
+};
+
+const uint8_t test_traits<BooleanType>::value(1);
+
+template <>
+struct test_traits<UInt8Type> {
+  static constexpr ParquetType::type parquet_enum = ParquetType::INT32;
+  static constexpr LogicalType::type logical_enum = LogicalType::UINT_8;
```

Comment:
What we might need to do is always return INT32 but provide an API where the user can request a particular column in the file to be casted to some other integer type.